src/interfaces/wallet.h

```diff
@@ -125,7 +125,7 @@ class Wallet
     virtual bool setAddressReceiveRequest(const CTxDestination& dest, const std::string& id, const std::string& value) = 0;
 
     //! Display address on external signer
-    virtual bool displayAddress(const CTxDestination& dest) = 0;
+    virtual std::optional<bilingual_str> displayAddress(const CTxDestination& dest) = 0;
```

Comment:
In 8218ce6b02b24eeef31f4e343ea777b7844907fc "wallet: return and display signer error"

Returning the error in an optional feels a bit wrong since it inverts are typical pattern. Rather than that the result is equivalent to `true`, we have to check that it's equivalent to `false`.

I think it would be better to use `util::Result<bool>` instead of an optional in order to preserve our typical error checking pattern.

Reply:
Done

Reply:
re: https://github.com/bitcoin/bitcoin/pull/24313#discussion_r1203035210

> Returning the error in an optional feels a bit wrong since it inverts the typical pattern. Rather than checking that the result is equivalent to `true`, we have to check that it's equivalent to `false`.

Returning `optional<bilingual_str>` or `Result<bool>` both seem ok here, but the ideal thing would be to return `Result<void>` so there are just 2 cases for calling code to handle 1-success, and 2-failure with an error message. Not 3 cases to handle: 1-success returning true, 2-success returning false, and 3-failure with an error message.

Complaining about the `optional<bilingual_str>` pattern is a recurring topic, see https://github.com/bitcoin/bitcoin/pull/27632#discussion_r1192007516, https://github.com/bitcoin/bitcoin/pull/27632#discussion_r1194858242, https://github.com/bitcoin/bitcoin/pull/25648#discussion_r935978206 and I have a PR to replace uses of `optional<bilingual_str>` with `util::Result` in #25977, but it's blocked waiting on a complicated base PR which adds support for `Result<void>`

Since this keeps coming up, though I think I will just add simple support for `Result<void>` to #25977 and decouple it from the base PR so it can get unstuck.

Just thinking out loud here, though (and advertising #25977)

Reply:
I switched to using util::Result in my last rebase a few months ago.